packages/cli-repl/src/cli-repl.ts

```diff
@@ -207,81 +218,11 @@ class CliRepl {
     });
   }
 
-  eval(originalEval: asyncRepl.OriginalEvalFunction, input: string, context: any, filename: string): Promise<any> {
-    this.lineByLineInput.enableBlockOnNewLine();
-    return this.shellEvaluator.customEval(originalEval, input, context, filename);
-  }
-
-  /**
-   * Creates a directory to store all mongosh logs, history and config
-   */
-  createMongoshDir(): void {
-    try {
-      mkdirp.sync(this.mongoshDir);
-    } catch (e) {
-      this._fatalError(e);
-    }
-  }
-
-  /**
-   * Checks if config file exists.
-   *
-   * If exists: sets userId and enabledTelemetry to this.
-   * If does not exist: writes a new file with a newly generated ObjectID for
-   * userid and enableTelemetry set to false.
-   */
-  generateOrReadTelemetryConfig(): void {
-    const configPath = path.join(this.mongoshDir, 'config');
-
-    let fd;
-
-    try {
-      fd = fs.openSync(configPath, 'wx');
-      this.userId = new bson.ObjectId().toString();
-      this.enableTelemetry = true;
-      this.disableGreetingMessage = false;
-      this.bus.emit('mongosh:new-user', this.userId, this.enableTelemetry);
-      this.writeConfigFileSync(configPath);
-    } catch (err) {
-      if (err.code === 'EEXIST') {
-        // make sure we catch errors for json parse and always have err and
-        // value on config result
-        let config: ConfigFileContents;
-        try {
-          config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
-        } catch (err) {
-          this.bus.emit('mongosh:error', 'Unable to parse user config', err);
-          return;
-        }
-
-        this.userId = config.userId;
-        this.disableGreetingMessage = true;
-        this.enableTelemetry = config.enableTelemetry;
-        this.bus.emit('mongosh:update-user', this.userId, this.enableTelemetry);
-        return;
-      }
-      this.bus.emit('mongosh:error', err);
-      throw err;
-    } finally {
-      if (fd !== undefined) fs.closeSync(fd);
-    }
-  }
-
-  /**
-   * sets CliRepl.enableTelemetry based on a bool, and writes the selection to
-   * config file.
-   *
-   * @param {boolean} enabled - enabled or disabled status
-   *
-   * @returns {string} Status of telemetry logging: disabled/enabled
-   */
-  toggleTelemetry(enabled: boolean): string {
-    this.enableTelemetry = enabled;
-    this.disableGreetingMessage = true;
-
-    this.bus.emit('mongosh:update-user', this.userId, this.enableTelemetry);
-    const configPath = path.join(this.mongoshDir, 'config');
-    this.writeConfigFileSync(configPath);
+  async toggleTelemetry(enabled: boolean): Promise<string> {
```

Comment:
I see why this may be convenient if it stays here, but it kinda seems this may even go somewhere else.

Would it be a bad idea to have a config that is just this interface:

```
interface Config {
    async isTelemetryEnabled(): boolean;
    async setTelemetryEnabled(boolean);
    async getHistoryPath();
    ...
}
```

Essentially abstracting away anything file system related or even the internal structure of the config to some more dedicated place?

Reply:
So … this particular function is being called from the shell evaluator, so I think we’re not really going to get around having a function that updates the config and triggers the store-to-filesystem operation in some way

But yeah, this is a good point, and I’ll think about it and either address it here or as part of MONGOSH-441 if I come up with something good

Reply:
I see so this is a different ticket. Got it.

`toggleTelemetry` probably could be called in the same way I guess in the shell evaluator. Anyway just leaving some random comment, all looks great!